src/__tests__/types.test.ts

```diff
@@ -598,4 +598,79 @@ describe('types', () => {
       });
     });
   });
+
+  describe('compound types', () => {
+    describe('or', () => {
+      test('default', () => {
+        const value = types.or(types.number({ required: true }), types.string());
```

Comment:
From an end-user perspective, what could it possibly mean for the `number` type to be required while the string type is not? It seems to me the `required` attribute should _not_ apply to the inner types but to the property that is `or` itself, no?

(This sort of reminds me of Array types and how we can set the inner type as optional, but that didn't make much sense, and was fixed in https://github.com/plexinc/papr/pull/231)